### ISSUE TITLE
Qute: include section - make it possible to supply the template id dynamically

### ISSUE THREAD

**Issue by mkouba** [quarkusio/quarkus#41050]
### Description

Currently, the first param (= the template id) is always treated as a string literal (i.e. it's not evaluated). For example, if there's a template that contains `{#include templateName /}` then Qute will attempt to lookup the template with id `templateName`.

### Implementation ideas

We cannot change the behavior of the `{#include}` section since it could break existing apps but we could either:

1. Introduce a new section (`{#includeDynamic}`?) that would treat the first param as an expression, or 
2. Introduce a config param (`_dynamic` or some better name?) that would be used to change the default behavior, i.e. something like `{#include templateName _dynamic /}`.

**Comment by myfear** [comment]
I'd love this being addressed. And here is a little use-case.

Dynamic UIs driven by AIServices.
I have created an agent that presents a list of "Entities" as results to the user. The list is returned in a json array and I want to include partials depending on the returned entity type to drive a dynamic UI for the end-user.

**Comment by mkouba** [comment]
@myfear what do you think about the `{#include foo.bar _dynamic /}` syntax/API? Maybe we could find a more accurate word than `_dynamic`?

**Comment by myfear** [comment]
What about:

{#include dynamic=true||false /} with the default / unset = false.

**Comment by mkouba** [comment]
> What about:
> 
> {#include dynamic=true||false /} with the default / unset = false.

Yes, it has to be disabled by default due to backward compatibility. 

But for the name we use the `_` prefix already (`_isolated`, `_ignoreFragments`, ...) in order to minimize conflict with a user-provided parameter.

So we could use `{#include foo.bar _dynamic=true /}`, or maybe something like `{#include foo.bar _dynamicLookup /}`, or even `{#include foo.bar _dynamicTemplate /}`.

**Comment by myfear** [comment]
In that case, I'm absolutely OK with _dynamic.

**Comment by mkouba** [comment]
> In that case, I'm absolutely OK with _dynamic.

I'd vote for `_dynamicTemplate` because it seems to be more accurate. I'll try to prepare a PR and we'll see ;-).

**Comment by FroMage** [comment]
Damnit, I know from the start that the syntax should have been `#{include 'foo.html'/}` 🤦

**Comment by mkouba** [comment]
> Damnit, I know from the start that the syntax should have been `#{include 'foo.html'/}` 🤦

Yup, we make mistakes, we learn... but we don't want to break existing stuff 🤷
